File: 1293-three-consecutive-odds/1293-three-consecutive-odds.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var threeConsecutiveOdds = function(arr) {
    let n = arr.length, mask = (1 << 1);
    let cnt = 0;
    for (let i = 0; i < n; ++i) {
        if (arr[i] & 1)
            cnt++;
        else
            cnt = 0;
        if ((cnt & 1) && (mask & cnt))
            return true;
    }
    return false;
};

File: 1293-three-consecutive-odds/1293-three-consecutive-odds.go
func threeConsecutiveOdds(arr []int) bool {
    cnt, mask := 0, 1 << 1
    for _, num := range arr {
        if num & 1 != 0 {
            cnt++
        } else {
            cnt = 0
        }

        if ((cnt & 1) != 0 && ((mask & cnt) != 0)) {
            return true
        }
    }
    return false
}